colcon_core/shell/sh.py

```diff
@@ -150,8 +151,13 @@ async def generate_command_environment(  # noqa: D102
             hook_path,
             {'dependencies': dependencies})
 
-        cmd = ['.', str(hook_path), '&&', 'env']
-        env = await get_environment_variables(cmd, cwd=str(build_base))
+        cmd = ['.', str(hook_path), '&&', 'env', '-0']
```

Comment:
Is `-0` added, so that https://github.com/colcon/colcon-core/pull/128 is not necessary any more? That PR already mentions the use of null as an alternative separator.

Hence, something like this:
> The patch won't do the right thing for multiline variables where the following lines still look like "key=value":
> ```
> FOO="bar
> BAZ=qux"
> ```

should work now?

Maybe add a comment here, why null-separated lines are requested here.

Reply:
This should indeed work around that limitation. It was the right call at the time: macOS didn't have any support for the `-0` flag. All currently supported macOS versions now support the flag, so it's a _fairly_ safe bet that we won't need to fall back to the existing implementation.

I'll add a comment.